Author: Tony Espy <espy@canonical.com>
Subject: Add new NMSettingsConnection reset_retries get/set methods

This patch adds get/set methods to NMSettingsConnection for the
reset_retries_timeout.  This allows sub-classes to override the
default setting ( 300s ).

Bug: https://bugs.launchpad.net/ubuntu/+source/network-manager/+bug/1461593

---
---
 src/settings/nm-settings-connection.c |   17 ++++++++++++++++-
 src/settings/nm-settings-connection.h |    3 +++
 2 files changed, 19 insertions(+), 1 deletion(-)

Index: b/src/settings/nm-settings-connection.c
===================================================================
--- a/src/settings/nm-settings-connection.c
+++ b/src/settings/nm-settings-connection.c
@@ -132,6 +132,7 @@ typedef struct {
 
 	int autoconnect_retries;
 	gint32 autoconnect_retry_time;
+	int reset_retries_timeout;
 	NMDeviceStateReason autoconnect_blocked_reason;
 
 	char *filename;
@@ -2103,6 +2104,19 @@ nm_settings_connection_get_autoconnect_r
 	return NM_SETTINGS_CONNECTION_GET_PRIVATE (connection)->autoconnect_retries;
 }
 
+int
+nm_settings_connection_get_reset_retries_timeout (NMSettingsConnection *connection)
+{
+	return NM_SETTINGS_CONNECTION_GET_PRIVATE (connection)->reset_retries_timeout;
+}
+
+void
+nm_settings_connection_set_reset_retries_timeout (NMSettingsConnection *connection,
+                                                  int timeout)
+{
+	NM_SETTINGS_CONNECTION_GET_PRIVATE (connection)->reset_retries_timeout = timeout;
+}
+
 void
 nm_settings_connection_set_autoconnect_retries (NMSettingsConnection *connection,
                                                 int retries)
@@ -2113,7 +2127,7 @@ nm_settings_connection_set_autoconnect_r
 	if (retries)
 		priv->autoconnect_retry_time = 0;
 	else
-		priv->autoconnect_retry_time = nm_utils_get_monotonic_timestamp_s () + AUTOCONNECT_RESET_RETRIES_TIMER;
+		priv->autoconnect_retry_time = nm_utils_get_monotonic_timestamp_s () + priv->reset_retries_timeout;
 }
 
 void
@@ -2277,6 +2291,7 @@ nm_settings_connection_init (NMSettingsC
 
 	priv->autoconnect_retries = AUTOCONNECT_RETRIES_DEFAULT;
 	priv->autoconnect_blocked_reason = NM_DEVICE_STATE_REASON_NONE;
+	priv->reset_retries_timeout = AUTOCONNECT_RESET_RETRIES_TIMER;
 
 	g_signal_connect (self, NM_CONNECTION_SECRETS_CLEARED, G_CALLBACK (secrets_cleared_cb), NULL);
 	g_signal_connect (self, NM_CONNECTION_CHANGED, G_CALLBACK (changed_cb), GUINT_TO_POINTER (TRUE));
Index: b/src/settings/nm-settings-connection.h
===================================================================
--- a/src/settings/nm-settings-connection.h
+++ b/src/settings/nm-settings-connection.h
@@ -195,6 +195,9 @@ void nm_settings_connection_reset_autoco
 
 gint32 nm_settings_connection_get_autoconnect_retry_time (NMSettingsConnection *connection);
 
+int nm_settings_connection_get_reset_retries_timeout (NMSettingsConnection *connection);
+void nm_settings_connection_set_reset_retries_timeout (NMSettingsConnection *connection, int timeout);
+
 NMDeviceStateReason nm_settings_connection_get_autoconnect_blocked_reason (NMSettingsConnection *connection);
 void nm_settings_connection_set_autoconnect_blocked_reason (NMSettingsConnection *connection,
                                                             NMDeviceStateReason reason);
